source/bdDetect.py

```diff
@@ -462,14 +462,14 @@ def getConnectedUsbDevicesForDriver(driver: str) -> Iterator[DeviceMatch]:
 			DeviceMatch(DeviceType.CUSTOM, port["usbID"], port["devicePath"], port)
 			for port in deviceInfoFetcher.usbDevices
 		),
-		(

```

Comment:
Could you please explain why you changed this?

Reply:
Currently, BrailleEdge products may simultaneously connect through Serial and HID connections. In cases where connection requests for both occur simultaneously, we aim to automatically prioritize and establish a Serial connection. Merely altering the sequence appeared to be without issues. If there are any objections or potential problems, your input would be greatly appreciated.

Reply:
Why would you prefer a USB serial connection over USB HID?

Reply:
@LeonarddeR 
The previous himsDriver did not use HID, so hims users are more familiar with Serial. This is why there were many opinions suggesting it's better to first explore Serial. Please let me know if there's any part you don't understand, and I would be grateful.

Reply:
I find it a bit tricky to adjust the logic in NVDA just for this. In ordering, HID has always been preferred over Serial. People might rely on this behavior and might get confused when we swap them.

Personally, I would say HID is the modern standard. Moreover, I assume that HID behaves exactly the same as serial? Or is HID a different choice in the device?

Reply:
Yes, that's correct. The device supports Serial in all countries, but HID is only supported in specific countries. There are countries where people who have been using Hims products traditionally choose the Serial method. If you know other solution, please tell me. I will follow.

Reply:
Perhaps we should discuss this in terms of user impact i.e. what issues your users are experiencing due to current ordering. I agree with @LeonarddeR that changing ordering for entire auto detection to accommodate a single Braille display driver seems a wrong choice.

Reply:
Thank you for your explanation.
And then is that impossible only change the ordering of detection in hims display?

Reply:
Could you please answer the first part of my question i.e. what is the user visible impact of the current ordering?

Reply:
There is a product that supports both HID and Serial connections. However, in certain countries, HID usage is not desired. In such cases, if the device automatically connects via HID first, users in countries where only Serial is used will be unable to use screen readers.

Reply:
> However, in certain countries, HID usage is not desired.

Are you saying that the Braille display technically supports HID, but users don't want to use it? If so I still have trouble understanding why they just cannot have it auto-detected using HID, and if desired switch to serial afterwards. Ideally if device exposes HID it should be as functional as serial is.

Reply:
Could you please answer this question from @lukaszgo1, @EdKweon ?

Reply:
@lukaszgo1 
In the case of serial connections, most functionalities can be operated without the NVDA screen reader. However, with HID connections, there is currently a total dependency on the NVDA screen reader. Due to this situation, users in certain countries have decided not to use HID. If implementing this code proves difficult, please let me know. I will roll back the code and explore other alternatives.